processor/schematransformerprocessor/config.go

```diff
@@ -0,0 +1,84 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schematransformerprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schematransformerprocessor"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/config"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schematransformerprocessor/internal/schema"
+)
+
+var (
+	errInvalidFileFormatVersion = errors.New("invalid file format version")
+	errInvalidSchemaURL         = errors.New("invalid URL schema")
+)
+
+// Config defines the user provided values for the Schema Transformer Processor
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"`
+
+	// SchemaURL defines where to download the published schema file
```

Comment:
The purpose of this field is unclear. Why does the user need to specify this? What will the processor do with this schema file?

Reply:
So my intentions for this field to be were to obtain the schema translation file of version `x`, then allow for semantic conversions from 1.0.0 to `x` (assuming that published version x is the most latest version to perform translations to, I will need to reread the spec to make sure).